Android Obb Extraction works

redwolf2 opened the pull request:
Obb extraction works. 
Next step: Import won't create the text assets in `/storage/emulated/0/Valkyrie/MoM/import/text`.

NPBruce commented on libraries/FFGAppImport/AssetImport/AppFinder.cs:
```diff
@@ -89,13 +91,11 @@ public AppFinder(Platform p)
-                {
-                    Directory.Delete(obbRoot, true);
-                }
+                //DeleteObb();
```
Won't this leave lots of data lying around?

NPBruce replied:
Actually this is on start.  This was here incase of a failed import/crash to clean up.  Unless there is a good reason I'd rather put it back in.

redwolf2 replied:
Went into problems with that, it's called multiple times. Maybe because of: 
`if (importData.platform == Platform.Android) return new MoMFinder(Platform.Android).RequiredFFGVersion(); // TODO: implement version checking for Android` in `fetchAppVerison()`. Will test it tomorrow.

redwolf2 commented:
I fixed the Obb deletion like you requested and added StreamingAssets import so `MoM/import/text` is available after import. The sources were in a different dir and had to be moved (Check `Appfinder.ConvertObbStreamingAssets`).

NPBruce commented on unity/Assets/Scripts/UI/Screens/GameSelectionScreen.cs:
```diff
@@ -1,6 +1,8 @@
 using UnityEngine;
 using FFGAppImport;
+using ValkyrieTools;
+using System;
```
I prefer not to have both System and UnityEngine because there are lots of conflicts.

redwolf2 commented:
Scenario can be started. Investigators choosen. But there are issues with the textures, preventing the scenario to be started for now.

NPBruce commented:
Great work!  I haven't had a chance to work this out yet: for streaming assets where are they placed and when are they removed?

redwolf2 commented:
They are within the obb file. Everything is extracted to

`/storage/emulated/0/Valkyrie`, which is the data directory like `%appdata%\Valkyrie`. Since /tmp can not be accessed in user mode, the App uses `/storage/emulated/0/Valkyrie/Obb` for the obb extraction and `/storage/emulated/0/Valkyrie/temp` fot the temp files.

The obb has the following folder structure:

```
assets
    ├─AssetBundles
    │  └─1.4.1
    │      └─Android
    │          ├─audio
    │          │  ├─coe
    │          │  ├─cosh
    │          │  ├─dd
    │          │  ├─dr
    │          │  ├─efi
    │          │  ├─general
    │          │  ├─gosm
    │          │  ├─rt
    │          │  ├─sb
    │          │  ├─vi
    │          │  └─wlw
    │          ├─localization
    │          └─textures
    │              └─logomad20
    └─bin
        └─Data
```

On extraction the Assets are in `assets/bin/Data` and the Streaming Assets are in the `asset/AssetBundles` structure. I copy them to the `assets/bin/Data` structure, so that they mirror the data folder structure on Windows.

The Obb dir is removed on every app start, when the AppFinder is initialized and should also be removed (not implemented yet), in the `Update()` method of the `GameSelectionScreen` (thats when the StreamingAssets were imported and the obb files are no longer needed).

NPBruce commented:
> The Obb dir is removed on every app start, when the AppFinder is initialized, on every reimport and should also be removed (not implemented yet), in the Update() method of the GameSelectionScreen (thats when the StreamingAssets were imported and the obb files are no longer needed, I left a TODO comment there.)

That's what I figured.  I will merge as-is for now, thanks.
